tests/installation/partitioning_smalldisk_storageng.pm

```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test to take the smallest disk in storage_ng scenarios
+#    Otherwise the partitioning proposal will use a free disk, which makes
+#    rebooting a game of chance on real hardware
+# Maintainer: Alvaro Carvajal <acarvajal@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+use version_utils qw(is_storage_ng);
+use partition_setup qw(take_first_disk);
+
+sub run {
+    if (!is_storage_ng) {
+        record_info "Requires storage_ng", "This module only works with storage_ng which is not present. Selecting first disk instead";
+        return take_first_disk;
+    }
+
+    select_console 'root-ssh';
```

Comment:
With this direct call this module can only be loaded on PVM or IPMI like backend, not qemu, right?

Could be a good idea to also use this test module for all HANA tests, WDYT? For another PR of course, as this is not the purposes of this one.

Reply:
I'll add a verification run over qemu just to confirm if this works, but I don't see this as being too useful for qemu. Usually in qemu tests we install the system in a test and then use the generated qcow2 image in another, and the second test is the one that has the extra disk. In that scenario we would not need this module (or partitioning_firstdisk for that matter).

Reply:
As suspected, this does not work over qemu: http://mango.qa.suse.de/tests/3203#step/partitioning_smalldisk_storageng/2

If you think it adds value to the module, I can add a conditional in place of the `select_console 'root-ssh'` so it uses `root-ssh` only on ikvm, ipmi, spvm or pvm_hmc, and `install-shell` or `root-console` on the other cases.

Reply:
if you think that this is not needed for qemu leave it as is. Maybe just add a comment in the header to notify that it's not intended to be used on qemu backend (you wrote about `real HW` but not sure that this is clear enough).

Reply:
It is working on qemu with `root-console`: http://mango.qa.suse.de/tests/3205

I had to create a needle for vda. Will do the same in textmode and push the changes to the code and the needles.

I don't know how useful this will be for qemu based tests, but this makes the module more general.

Reply:
qemu test in textmode: http://mango.qa.suse.de/tests/3215